include/cetl/any.hpp

```diff
@@ -26,69 +26,302 @@ class any;
+            const auto ptr = static_cast<Tp*>(storage);
+            ptr->~Tp();
+        };
+    }
+
+    template <typename ValueType>
+    CETL_NODISCARD void* get_ptr() noexcept
+    {
+        if (!has_value())
+        {
+            return nullptr;
+        }
+
+        // TODO: Add RTTI check here.
+        return get_raw_storage();
+    }
+
+    template <typename ValueType>
+    CETL_NODISCARD const void* get_ptr() const noexcept
+    {
+        if (!has_value())
+        {
+            return nullptr;
+        }
+
+        // TODO: Add RTTI check here.
+        return get_raw_storage();
+    }
+
+    void copy_handlers_from(const base_storage& src) noexcept
+    {
+        value_destroyer_ = src.value_destroyer_;
+    }
+
+    void reset() noexcept
+    {
+        if (value_destroyer_)
+        {
+            value_destroyer_(get_raw_storage());
+            value_destroyer_ = nullptr;
+        }
+    }
+
+private:
     // We need to align the buffer to the given value (maximum alignment by default).
     // Also, we need to ensure that the buffer is at least 1 byte long.
     alignas(Alignment) char buffer_[std::max(Footprint, 1UL)];
-};
 
-// Movable case.
-template <std::size_t Footprint, bool Copyable, std::size_t Alignment>
-struct base_move<Footprint, Copyable, true, Alignment> : base_copy<Footprint, Copyable, Alignment>
-{
-public:
-    constexpr base_move()                      = default;
-    base_move(base_move&&) noexcept            = default;
-    base_move& operator=(base_move&&) noexcept = default;
-};
+    // Holds type-erased value destroyer. `nullptr` if storage has no value stored.
+    void (*value_destroyer_)(void* self) = nullptr;
 
-// Non-movable case.
-template <std::size_t Footprint, bool Copyable, std::size_t Alignment>
-struct base_move<Footprint, Copyable, false, Alignment> : base_copy<Footprint, Copyable, Alignment>
+};  // base_storage
+
+//
+//
+template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment>
+struct base_handlers;
+//
+template <std::size_t Footprint, std::size_t Alignment>
+struct base_handlers<Footprint, false, false, Alignment> : base_storage<Footprint, Alignment>
+{};
+//
+template <std::size_t Footprint, std::size_t Alignment>
+struct base_handlers<Footprint, true, false, Alignment> : base_storage<Footprint, Alignment>
 {
-public:
-    constexpr base_move()                      = default;
-    base_move(base_move&&) noexcept            = delete;
-    base_move& operator=(base_move&&) noexcept = delete;
-};
+    void copy_handlers_from(const base_handlers& src) noexcept
+    {
+        base::copy_handlers_from(src);
+        value_copier_ = src.value_copier_;
+    }
 
-// Copyable case.
+    void reset() noexcept
+    {
+        value_copier_ = nullptr;
+        base::reset();
+    }
+
+    // Holds type-erased value copyer. `nullptr` when copy operation is not supported.
+    void (*value_copier_)(const void* src, void* dst) = nullptr;
+
+private:
```

Comment:
Here and below, I don't really see the point of this specifier; I think it should be removed.